erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -251,6 +251,17 @@ frappe.ui.form.on('Stock Entry', {
 		frm.trigger("setup_quality_inspection");
 	},
 
+	stock_entry_type: function(frm){
+		frm.remove_custom_button('Bill of Materials', "Get items from");
+
+		if (frm.doc.docstatus === 0 && ['Material Issue','Material Receipt',
+		'Material Transfer','Send to Subcontractor'].includes(frm.doc.purpose)) {
```

Comment:
use standard in_list method

Reply:
in_list is a wrapper around list.includes(item), isn't any difference. I formatted it better in the latest commit